src/engine.vala

```diff
@@ -20,8 +20,37 @@
 using Gee;
 
 class KkcEngine : IBus.Engine {
-    // Preferences are shared among KkcEngine instances.
-    static Preferences preferences;
+    public bool show_annotation { get; set; }
+    public bool use_custom_keymap { get; set; }
+    public IBus.Keymap keymap { get; set; }
+    public uint page_size { 
+        set {
+            // lookup_table.page_size is not a property and set the page_size
+            // to lookup_table.page_size here.
+            context.candidates.page_size = (uint)value;
+            lookup_table.set_page_size (value);
+        }
+    }
+    public int punctuation_style {
+        set {
+            // Keep "punctuation-style" key as int and convert int to
+            // enum here because Setup.punctuation_style_combobox
+            // does not have an enum property for Settings.bind().
+            context.punctuation_style = (Kkc.PunctuationStyle)value;
```

Comment:
Again, read the documentation before adding such ugly hack:
https://developer.gnome.org/gio/stable/GSettings.html
(search for "enumerated types").

Reply:
I meant Variant in Settings.bind() does not have getter for enum property but not gsettings_get_enum_value(). I have already read that pointer before I request the migration.
But from your reply, I guessed you might know that treatment in Settings.bind()  and investigated it today.